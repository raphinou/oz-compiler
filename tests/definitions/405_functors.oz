% functor without prepare
local
   F=functor
      export
         echo:Echo
      import
         System(show:ImportedShow)
      define
         {ImportedShow 'define::start'}
         proc {Echo S}
            {PrivateEcho S}
         end
         proc {PrivateEcho S}
            {ImportedShow S}
         end
         {ImportedShow 'define::end'}
   end
   M
in
   M={Module.apply [F]}.1
   {M.echo 'hello'}
end

