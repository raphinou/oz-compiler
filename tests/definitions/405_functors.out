'define::start'
'define::end'
hello
